ext/mysql2/client.c

```diff
@@ -57,6 +57,16 @@ static ID intern_brackets, intern_merge, intern_merge_bang, intern_new_with_args
   #define MYSQL_LINK_VERSION MYSQL_SERVER_VERSION
 #endif
 
+/*
+ * mariadb-connector-c defines CLIENT_SESSION_TRACKING
+ * while mysql-connector-c defines CLIENT_SESSION_TRACK.
+ * This is a hack to take care of both clients.
+ */
+#if defined(CLIENT_SESSION_TRACK)
+#elif defined(CLIENT_SESSION_TRACKING)
+  #define CLIENT_SESSION_TRACK CLIENT_SESSION_TRACKING
+#endif
```

Comment:
Similar hacks are used in few lines above in https://github.com/Shopify/mysql2/blob/360e63d51c282658131777e7b5519639294dd7e4/ext/mysql2/client.c#L52-L55